Gems/PhysX/Code/Tests/CharacterGameplayControllerTests.cpp

```diff
@@ -114,41 +117,136 @@ namespace PhysX
         int duration = 10;
-
         for (int i = 0; i < duration; i++)
         {
             basis.m_gameplayController->SetGravityMultiplier(expectedGravityMultiplier + i);
             basis.Update();
 
-            EXPECT_THAT(expectedGravityMultiplier + i, testing::FloatEq(basis.m_gameplayController->GetGravityMultiplier()));
+            EXPECT_THAT(basis.m_gameplayController->GetGravityMultiplier(), testing::FloatEq(expectedGravityMultiplier + i));
+        }
+    }
+
+    //! Test that verifies that the Falling Velocity Set & Get methods function as expected.
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_FallingVelocitySets)
+    {
+        const AZ::Vector3 expectedVelocity(0.0f, 0.0f, 22.0f);
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        // Get the original velocity so we can determine that set actually changes things.
+        auto originalVelocity = basis.m_gameplayController->GetFallingVelocity();
+
+        // Set the falling velocity to the expected + original velocity to verify that the velociy will change.
+        basis.m_gameplayController->SetFallingVelocity(originalVelocity + expectedVelocity);
+
+        // Get the end velocity and validate that it is the expected value.
+        auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+        EXPECT_THAT(endVelocity.GetZ(), testing::FloatNear(expectedVelocity.GetZ(), 0.001f));
+    }
+
+    //! Test the verifies that the Falling Velocity sets and uses the set value to interact with the tick incrementing.
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_FallingVelocitySetsWhileMoving)
+    {
+        const AZ::Vector3 expectedVelocity(0.0f, 0.0f, 22.0f);
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        // Get the original velocity so we can determine that set actually changes things.
+        auto originalVelocity = basis.m_gameplayController->GetFallingVelocity();
+
+        // Run the scene for a while and validate that the set velocity is working
+        for (int i = 0; i < 10; i++)
+        {
+            // Calculate and set velocity based upon the original velocity, expected velocity, and loop count modifier
+            auto setVelocity = originalVelocity + expectedVelocity + AZ::Vector3(0.0f, 0.0f, float(i));
+            basis.m_gameplayController->SetFallingVelocity(setVelocity);
+
+            basis.Update();
+
+            // Get the endVelocity & current gravity to perform calculations on.
+            auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+            const auto gravity = basis.m_testScene->GetGravity();
+
+            // The end velocity should be decelerated by gt when g is gravity acceleration and t is the length of time
+            const float expectedDeceleration = gravity.GetZ() * basis.m_timeStep;
```

Comment:
Nitpick - I would call this something like expectedVelocityChange. The name expectedDeceleration implies that it is a rate of change of velocity, which is slightly different.

Reply:
updated